### PR TITLE
Create EvaluationQueueProcessor base class, and validator/scorer classes

### PR DESCRIPTION
**Pull request by thomasyu888**
Created ABCMeta class for evaluation queue processor, as the workflow is usually, download submissions with `x` status, interact with these submissions, update submission status, notify submitter.

### CODE REVIEW

File: scoring_harness/helper.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+                                                     to_public=True)
+        sub_status.status = self._success_status if submission_info['valid'] else "INVALID"
+
+        if not self.dry_run:
+            sub_status = self.syn.store(sub_status)
+
+    def notify(self, submission, submission_info):
+        """Notify submitter or admin"""
+
+    def pipeline(self, status):
```
Comment (kdaily) [inline scoring_harness/helper.py]: change name to something more generic

Comment (kdaily) [reply scoring_harness/helper.py]: or overload `__call__`

File: scoring_harness/validater.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import logging
+from .helper import EvaluationQueuePipeline
+from . import messages
+
+logging.basicConfig(format='%(asctime)s %(message)s')
+LOGGER = logging.getLogger(__name__)
+LOGGER.setLevel(logging.INFO)
+
+
+class EvaluationQueueValidater(EvaluationQueuePipeline):
```
Comment (kdaily) [inline scoring_harness/validater.py]: `Validator`

File: bin/runchallenge.py [hunk 3]
```diff
@@ -37,52 +36,39 @@ def import_config_py(config_path):
...
-        syn.login(email=args.user, password=args.password, silent=True)
+        if args.synapse_config is not None:
+            syn = synapseclient.Synapse(debug=args.debug,
+                                        configPath=args.synapse_config)
+        else:
+            syn = synapseclient.Synapse(debug=args.debug)
+        syn.login(silent=True)
     except (SynapseAuthenticationError, SynapseNoCredentialsError):
         raise ValueError(
             "Must provide Synapse credentials as parameters or "
```
Comment (thomasyu888) [inline bin/runchallenge.py]: Update value error.  Don't store credentials are env variables.

**Pull request merged**